src/main/java/staffconnect/model/meeting/MeetingList.java

```diff
@@ -0,0 +1,119 @@
+ * Supports a minimal set of list operations.
+ */
+public class MeetingList implements Iterable<Meeting> {
+
+    private final ObservableList<Meeting> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Meeting> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the list contains an equivalent meeting as the given argument.
+     */
+    public boolean contains(Meeting toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::equals);
+    }
+
+    /**
+     * Adds a meeting to the list.
+     * The meeting must not already exist in the list.
+     */
+    public void add(Meeting toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateMeetingException();
+        }
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Removes the equivalent meeting from the list.
+     * The meeting must exist in the list.
+     */
+    public void remove(Meeting toRemove) {
+        requireNonNull(toRemove);
+        if (!internalList.remove(toRemove)) {
+            throw new MeetingNotFoundException();
+        }
+    }
+
+    /**
+     * Replaces the contents of this list with {@code meeting}.
+     * {@code meetings} must not contain duplicate meetings.
+     *
+     */
+    public void setMeetings(List<Meeting> meetings) {
+        requireAllNonNull(meetings);
+        if (!meetingsAreUnique((meetings))) {
+            throw new DuplicateMeetingException();
+        }
+
+        internalList.setAll(meetings);
+    }
+
+    /**
+     * Returns the backing list as an unmodifiable {@code ObservableList}.
+     */
+    public ObservableList<Meeting> asUnmodifiableObservableList() {
+        return internalUnmodifiableList;
+    }
+
+    @Override
+    public Iterator<Meeting> iterator() {
+        return internalList.iterator();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof MeetingList)) {
+            return false;
+        }
+
+        MeetingList otherMeetingList = (MeetingList) other;
+        return internalList.equals(otherMeetingList.internalList);
+    }
+
+    @Override
+    public int hashCode() {
+        return internalList.hashCode();
+    }
+
+    @Override
+    public String toString() {
+        return internalList.toString();
+    }
+
+    /**
+     * Returns true if {@code meetings} contains only unique Meeting.
+     */
+    private boolean meetingsAreUnique(List<Meeting> meetings) {
+        for (int i = 0; i < meetings.size() - 1; i++) {
+            for (int j = i + 1; j < meetings.size(); j++) {
+                if (meetings.get(i).equals(meetings.get(j))) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
```

Comment:
I think this is actually fine because it fits 3 levels of indentations, as the rule of thumb was to have not more than 3 . Also there is a similar case of this is in ```UniquePersonsList```. Of course I can change the implementation to utilise functional programming instead.

Reply:
oh then it should be fine to leave it as is.
since `UniquePersonsList` uses the same implementation, perhaps we can decide on whether to change the implementation at a later time.